### PR TITLE
feat: add github action "terraform"

### PR DESCRIPTION
**Pull request by keroro520**
Action desctiption: [action-terraform](https://github.com/keroro520/ckb/blob/action-terraform/.github/actions/terraform/README.md)

CI tests "benchmark" and "sync-mainnet" apply AWS instances on-demand. I created a docker action using the Github Action `post` keyword to ensure that the applied instances must be destroyed.

### CODE REVIEW

**Comment by nervos-bot-user** [comment]
Can one of the admins verify this patch?

**Review by quake — APPROVED** [review]

File: .github/actions/terraform/main.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# Path to location of terraform files
+TERRAFORM_DIR=${INPUT_TERRAFORM_DIR:-${TERRAFORM_DIR}}
```
Comment (keroro520) [inline .github/actions/terraform/main.sh:6]: Github Actions transform the [action input `terraform_dir`](https://github.com/nervosnetwork/ckb/pull/2679/files#diff-60065d341a47aaac2c5285ccaf4bc533609632cf74aa7603f80b1de0f2e37ac2R6) into environment variable `INPUT_TERRAFORM_DIR`.

**Comment by doitian** [comment]
Is there a passed example?

File: .github/actions/terraform/apply.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+# Call by action `entrypoint`
+
+SCRIPT_PATH="$( cd -- "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
```
Comment (doitian) [inline .github/actions/terraform/apply.sh:5]: You can set `WORKDIR /terraform` in the Dockerfile

**Review by doitian — APPROVED** [review]

File: .github/actions/terraform/apply.sh (hunk 2 above)
Comment (keroro520) [reply .github/actions/terraform/apply.sh:5]: https://docs.github.com/en/actions/creating-actions/dockerfile-support-for-github-actions#workdir

> GitHub sets the working directory path in the GITHUB_WORKSPACE environment variable. It's recommended to not use the WORKDIR instruction in your Dockerfile.

**Pull request merged**